base/src/java/lang/util.d

```diff
@@ -385,33 +385,187 @@ version(Tango){
                 Formatter( "{}", f ) == "{3.14 => PI, 1.00 => one}");*/
     }
 
-    class Format{
+    /// assitive enum for `Format`.
+    private enum ArgType {

```

Comment:
Since this is only used inside the `Format` class, you can put it there. In theory it could be placed in the `opCall` method. But because it's a template it will cause an explosion of symbols and type info.

Reply:
Fixed. Didn't know that an `enum` could be declared in a class.

Reply:
All declarations can be placed at any scope in D, except for the module decoration. So called "turtles all the way down".